### PR TITLE
Add an example demonstrating the use of the invoke_function_action facility

### PR DESCRIPTION
**Pull request by hkaiser**
- also applies fixes to the facility itself.

### CODE REVIEW

**Review by msimberg — DISMISSED** [review]

File: examples/quickstart/invoke_function.cpp [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+        std::cout << "the action invocation returned: " << result.get() << "\n";
+    }
+
+    {
+        using action_type = hpx::components::server::invoke_function_action<
+            decltype(&void_call_me), int>;
+
+        hpx::future<void> result = hpx::async<action_type>(hpx::find_here(),
+            reinterpret_cast<std::ptrdiff_t>(&void_call_me), 42);
```
Comment (aurianer) [inline examples/quickstart/invoke_function.cpp:46]: Can you explain why do you use `ptrdiff_t`? I don't know this well..

**Review by aurianer — DISMISSED** [review]
Thanks!

File: examples/quickstart/invoke_function.cpp (hunk 1 above)
Comment (hkaiser) [reply examples/quickstart/invoke_function.cpp:46]: `std::size_t` would have worked as well. Both types are guaranteed to be large enough integral types to be used for roundtrip casting pointer->integral->pointer.

Comment (hkaiser) [reply examples/quickstart/invoke_function.cpp:46]: However it looks like that gcc and clang are not happy with this choice. I'll change this to `std::size_t`.

**Comment by gonidelis** [comment]
Late comment: Thanks for that!

**Pull request merged**